lib/dao.js

```diff
@@ -291,6 +292,11 @@ DataAccessObject.create = function (data, options, cb) {
               }
               var context = {
                 Model: Model,
+                /* TODO (pradnya), here obj, does NOT have the attributes, 
+                 * newly added/updated through persist hook. how should 
+                 * `obj` be updated, so that the following test succeeds
+                 * 'applies updates from `persist` hook - on create'
+                 */
```

Comment:
I think you should make two changes:
1) In the test, add another check to verify what has been stored in the database. I believe this check should be already passing with the current implementation.
2) To fix the particular issue you have described, you will need to copy properties from `data` back to the instance. I think `obj.setAttributes(context.data);` should do the job, see the implementation of `updateAttributes` which calls that method too.